ipyvolume/serialize.py

```diff
@@ -106,6 +106,56 @@ def cube_to_png(grid, vmin, vmax, file):
 		img.save(file, "png")
 	return (image_width, image_height), tile_shape, rows, columns, slices
 
+def tile_volume(vol, tex_size, nr_of_tiles, vol_size):
+	# now tiling is always square, if volume is for example a x/y ratio of 2/1 it will create a big texture
+	# which will only be filled for half, needs to be changed based on ratio of x/y
+	tex = np.zeros(tex_size,dtype=vol.dtype)
+	for tileY in range(nr_of_tiles[1]):
+	    for tileX in range(nr_of_tiles[0]):
+	        z = tileX + tileY * nr_of_tiles[0]
+	        if z >= vol_size[2]:
+	            break
+	        slice_data = vol[z]
+	        xoffset = tileX*vol_size[0]
+	        yoffset = tileY*vol_size[1]	
+	        tex[yoffset:yoffset+vol_size[1],xoffset:xoffset+vol_size[0]] = slice_data
+	# debug image saving
+	# scipy.misc.toimage(tex, cmin=tex.min(), cmax=tex.max()).save('outfile.png')
+
+
+	return array_to_binary(tex)
+
+
+def volume_to_json_volume_tiled(vol, obj=None):
+	if vol is None:
+		return None
+	vol = np.asarray(vol)
+
+	# Keeping things square
+	# x * y * a^2 = z
+	# a = sqrt(z/(x*y))
+	# nrtilex = y*a
+	# nrtiley = x*a
+	vol_size = vol.shape[-3:][::-1]
```

Comment:
Yea wouldn't want to call it vol.shape because that is mostly zyx and I wanted the shape to be naturally xyz. Because I see it as xyz resolution size for a volume. It is probably what your used to but I'll call it volume_shape then.